### DISCUSSION TITLE
Change material of one object by clicking another?

### DISCUSSION THREAD

**Discussion by phatpixels** [google/model-viewer#4379] [Q&A] [answered]
Hi All,  
Iam trying to alter the material of one object by clicking on another. Right now I can alter the same object I'm clicking on, but I need to target another objects material, I tried using this `mv.model.getMaterialByName('name')` but I'm not sure how to construct the rest. I'm more of a designer practicing code, any small snippet of code to get me going on this would be greatly appreciated.

**Comment by danilokrulj** [comment, marked as answer]
You use that to define the material you are trying to change.
const mat1 = modelViewer.model.getMaterialByName('name');

Than you need to create textures from the images you wish to use.
const createTextures = async () => {
      texture1 = await modelViewer.createTexture("assets/mat1_Base_Color2.jpg");
      texture2 = await modelViewer.createTexture("assets/mat2_Base_Color1.jpg");
    }
    
Than you need to apply those textures to the material.
mat1.pbrMetallicRoughness.baseColorTexture.setTexture(texture1);

Or change the baseColor of the material.
mat1.pbrMetallicRoughness.setBaseColorFactor([0.5,0.065,0.043,1]);

And so on. Hope this helps you a little.

**Reply by phatpixels** [reply]
@danilokrulj thankyou kindly this has got me on the right path to achieve what i needed :-)
